CSharpMath/Structures/Dictionary.cs

```diff
@@ -0,0 +1,278 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+
+namespace CSharpMath.Structures {
+
+  [SuppressMessage("Naming", "CA1710:Identifiers should have correct suffix", Justification = NotACollection)]
+  [SuppressMessage("Design", "CA1010:Collections should implement generic interface", Justification = NotACollection)]
+  public class ProxyAdder<TKey, TValue> : IEnumerable {
+    const string NotACollection = "This is not a collection. It implements IEnumerable just to support collection initializers.";
+    [Obsolete(NotACollection, true)]
+    [SuppressMessage("Design", "CA1033:Interface methods should be callable by child types", Justification = NotACollection)]
+    IEnumerator IEnumerable.GetEnumerator() => throw new NotSupportedException(NotACollection);
```

Comment:
Should not support IEnumerable and then fail on the implementation.

Reply:
Using collection initializers in types not implementing `IEnumerable` will result in errors despite having reachable `Add` methods.